### PR TITLE
fix: Remove "/" from crash report file name

### LINKED ISSUE #2975: No automatic Crash upload
### Platform

iOS

### Installed

CocoaPods

### Version

8.5.0

### Steps to Reproduce

1. First, I registered the SentrySDK in the first line of the `application(_:, didFinishLaunchingWithOptions:)` method:
```swift
public func start(environment: String? = nil, release: String? = nil) {
    guard !SentrySDK.isEnabled else { return }

    SentrySDK.start {
        $0.dsn = Config.sentry.dsn
        $0.debug = !Bool.isRelease
        $0.attachScreenshot = true
        $0.attachViewHierarchy = true
        $0.releaseName = release ?? Bundle.sak.shortVersionString()
        $0.dist = Bundle.sak.bundleVersionString()
        
        if let environment = environment {
            $0.environment = environment
        }
     }
}
```

2. Running App on iPhone XS Max with 14.7.1.
3. Click a pre-set button that executes the `SentrySDK.crash()` method after the button is clicked.

### Expected Result

App crashes and receives a crash log in our self-built Sentry panel.

### Actual Result

The app crashes, but the web panel is empty.

--------

I've tried calling the `capture()` method and manually reporting an `NSError`, and it shows up immediately in the web panel.

I also get `WatchdogTermination` in the web panel, but the only thing that doesn't show up is the crash.

Are there any configurations that would affect this behavior? Any help would be appreciated, thanks.

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Crash report uses bundleName to create the report file, but "/" is a valid character for bundle name but not for file name.
Replace "/" with "-" for the report file name.

## :bulb: Motivation and Context

Close #2975

## :green_heart: How did you test it?

Unit test and sample.

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/SentryCrash/Recording/SentryCrash.m [hunk 1]
```diff
@@ -67,10 +67,19 @@
...
+    static NSString *bundleName = nil;
+
     if (bundleName == nil) {
-        bundleName = @"Unknown";
+#if TEST
+        bundleName = @"Sentry/Test";
+#else
+        bundleName =
+            [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleName"] ?: @"Unknown";
+#endif
```
Comment (philipphofmann) [inline Sources/SentryCrash/Recording/SentryCrash.m:78]: `m`: I would prefer finding a different way of testing this. One way would be to define an extra method or class that handles the edge case below and directly call it in tests.

File: Sources/SentryCrash/Recording/SentryCrash.m [hunk 2]
```diff
@@ -67,10 +67,19 @@
 static NSString *
 getBundleName(void)
 {
-    NSString *bundleName = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleName"];
+    static NSString *bundleName = nil;
+
     if (bundleName == nil) {
-        bundleName = @"Unknown";
+#if TEST
+        bundleName = @"Sentry/Test";
```
Comment (rakuyoMo) [inline Sources/SentryCrash/Recording/SentryCrash.m:74]: I think this is very dangerous. In many componentized projects, developers bring some macro definitions to pod components, and TEST is a very common one, used as a marker in test environments. In this case the bundleName will always be "Sentry/Test"

Comment (brustolin) [reply Sources/SentryCrash/Recording/SentryCrash.m:74]: Although I believe a dev should not add a macro definition to a external library pod this is a fair point, but we do have other places with TEST in the project.
We probably should rethink this.

Comment (rakuyoMo) [reply Sources/SentryCrash/Recording/SentryCrash.m:74]: > Although I believe a dev should not add a macro definition to a external library pod this is a fair point, but we do have other places with TEST in the project.
> We probably should rethink this.

In a componentized project, although we can use some hard coding to distinguish which are business components and which are third-party components, some inexperienced developers may ignore this and bring macro definitions to all pods.

I think the right approach is that the third-party pod and the main app each ensure the uniqueness of their own macro definitions.

File: Sources/SentryCrash/Recording/SentryCrash.m [hunk 3]
```diff
@@ -562,6 +534,35 @@ - (void)applicationWillTerminate
     sentrycrash_notifyAppTerminate();
 }
 
+- (NSString *)clearBundleName:(NSString *)filename
+{
+    return [filename stringByReplacingOccurrencesOfString:@"/" withString:@"-"];
```
Comment (philipphofmann) [inline Sources/SentryCrash/Recording/SentryCrash.m:539]: `m`: Please briefly comment on why we added this logic.

**Review by philipphofmann — APPROVED** [review]
Thanks, @brustolin.

**Comment by armcknight** [comment]
Where can a valid bundle ID be made with a forward slash? ADP just won't allow it (screenshot 1), and Xcode automatically converts my input (screenshot 2) to a hyphen after pressing enter (screenshot 3), and that change is reflected in the xcodeproj (screenshot 4): 
image
image
image
image

File: Sources/SentryCrash/Recording/SentryCrash.m (hunk 3 above)
Comment (brustolin) [inline Sources/SentryCrash/Recording/SentryCrash.m:539]:
```suggestion
    //The bundle name is used as file name, therefore "/" is not allowed.
    return [filename stringByReplacingOccurrencesOfString:@"/" withString:@"-"];
```

**Pull request merged**